tests/utils/test_env.py

```diff
@@ -876,7 +876,7 @@ def err_on_rm_venv_only(path: Path | str, *args: Any, **kwargs: Any) -> None:
             remove_directory(path)
 
     m = mocker.patch(
-        "poetry.utils.env.remove_directory", side_effect=err_on_rm_venv_only
```

Comment:
This should probably just mock `poetry.utils.helpers.remove_directory` as that is where the method actually comes from.

Reply:
I have tried that, but if I do so, the test fails with `FAILED tests/utils/test_env.py::test_remove_keeps_dir_if_not_deleteable - AssertionError: remove_directory(PosixPath('/tmp/poetry_1s94p2i5/simple-project-qCH79UIk-py3.6')) call not found`, didn't really know how to properly debug this, so decided to leave it as it is.